app/src/main/java/org/stepik/android/view/onboarding/resolver/OnboardingRemoteConfigResolver.kt

```diff
@@ -3,18 +3,160 @@ package org.stepik.android.view.onboarding.resolver
 import com.google.firebase.remoteconfig.FirebaseRemoteConfig
 import com.google.gson.Gson
 import com.google.gson.reflect.TypeToken
+import org.stepic.droid.analytic.Analytic
 import org.stepic.droid.configuration.RemoteConfig
+import org.stepic.droid.preferences.SharedPreferenceHelper
+import org.stepik.android.domain.onboarding.analytic.OnboardingParseErrorAnalyticEvent
 import org.stepik.android.view.onboarding.model.OnboardingGoal
 import javax.inject.Inject
 
 class OnboardingRemoteConfigResolver
 @Inject
 constructor(
+    private val analytic: Analytic,
+    private val sharedPreferenceHelper: SharedPreferenceHelper,
     private val firebaseRemoteConfig: FirebaseRemoteConfig,
     private val gson: Gson
 ) {
     fun buildOnboardingGoals(): List<OnboardingGoal> {
         val onboardingGoalsString = firebaseRemoteConfig.getString(RemoteConfig.PERSONALIZED_ONBOARDING_COURSE_LISTS)
-        return gson.fromJson(onboardingGoalsString, TypeToken.getParameterized(ArrayList::class.java, OnboardingGoal::class.java).type)
+
+        return try {
+            gson.fromJson(onboardingGoalsString, TypeToken.getParameterized(ArrayList::class.java, OnboardingGoal::class.java).type)
+        } catch (e: Exception) {
+            analytic.report(OnboardingParseErrorAnalyticEvent(onboardingGoalsString, sharedPreferenceHelper.isOnboardingNotPassedYet))
```

Comment:
Does `sharedPreferenceHelper.isOnboardingNotPassedYet` mean `firstTime`?

Reply:
Decided to use `sharedPreferenceHelper.lastSessionTimestamp` instead